src/components/flyout/flyout_header.styles.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { css } from '@emotion/react';
+import { UseEuiTheme } from '../../services';
+import { logicalCSS } from '../../global_styling';
+
+export const euiFlyoutHeaderStyles = (euiThemeContext: UseEuiTheme) => {
+  const euiTheme = euiThemeContext.euiTheme;
+
+  return {
+    euiFlyoutHeader: css`
+      flex-grow: 0;
+    `,
+    'flyoutHeader--hasBorder': css`
```

Comment:
I'd just name this `hasBorder` personally since Emotion concatenates the class names for us and will output `euiFlyoutHeader-hasBorder`
```suggestion
    'hasBorder': css`
```